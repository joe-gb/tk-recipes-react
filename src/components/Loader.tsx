import React from "react";
import styled from "styled-components";

const Loader = styled.div`
  text-align: center;
  box-sizing: border-box;
  margin: 0 auto;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  border: 4px solid;
  border-color: #144fcc #144fcc rgba(20, 79, 204, 0.30000000000000004)
    rgba(20, 79, 204, 0.30000000000000004);
  -webkit-animation: spin 0.8s infinite linear;
  animation: spin 0.8s infinite linear;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export default Loader;
