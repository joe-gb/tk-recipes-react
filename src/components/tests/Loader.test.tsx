import React from "react";
import { render } from "@testing-library/react";
import Loader from "../Loader";

test("Loader snapshot test", () => {
  let { container } = render(<Loader />);
  expect(container).toMatchSnapshot();
});
