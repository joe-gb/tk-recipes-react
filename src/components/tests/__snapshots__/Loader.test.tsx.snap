// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot test 1`] = `
.c0 {
  text-align: center;
  box-sizing: border-box;
  margin: 0 auto;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  border: 4px solid;
  border-color: #144fcc #144fcc rgba(20,79,204,0.30000000000000004) rgba(20,79,204,0.30000000000000004);
  -webkit-animation: spin 0.8s infinite linear;
  -webkit-animation: spin 0.8s infinite linear;
  animation: spin 0.8s infinite linear;
}

<div>
  <div
    class="c0"
  />
</div>
`;
